azure/types.go

```diff
@@ -372,6 +375,12 @@ type AADProfile struct {
 	AdminGroupObjectIDs []string
 }
 
+// SKU - AKS SKU.
+type SKU struct {
+	// Tier - Tier of a managed cluster SKU.
+	Tier string
+}
+
```

Comment:
```suggestion

Reply:
because we need only Tier information from the SKU struct, can we just add it in the ManagedClusterSpec and avoid using a separate struct for this ?

Reply:
I’m pretty neutral on this. The current approach aligns to the AKS api, which was intended to be extensible, but we also don’t have any plans to extend it currently.

Reply:
any possibility that it can be extended in the future ??

Reply:
It can change from AKS side yes. If AKS changes it in the future, we will have to make a breaking change to align it to the impl already in this PR

Reply:
yeah so I think it is better to not change then 👍

Reply:
@alexeldeib apart from the above changes this PR lgtm :) Great work @richardchen331